src/components/datagrid/body/header/data_grid_header_cell_wrapper.tsx

```diff
@@ -49,20 +49,26 @@ export const EuiDataGridHeaderCellWrapper: FunctionComponent<
   }, [index, setFocusedCell]);
 
   const [isFocused, setIsFocused] = useState(false);
-  useEffect(() => {
-    onFocusUpdate([index, -1], (isFocused: boolean) => {
-      setIsFocused(isFocused);
-    });
-  }, [index, onFocusUpdate]);
 
-  useEffect(() => {
+  const updateFocus = (isFocused: Boolean, headerEl: HTMLDivElement | null) => {
     if (isFocused && headerEl) {
       // Only focus the cell if not already focused on something in the cell
       if (!headerEl.contains(document.activeElement)) {
         headerEl.focus();
       }
     }
-  }, [isFocused, headerEl]);
+  }
+
+  useEffect(() => {
+    onFocusUpdate([index, -1], (isFocused: boolean) => {
+      setIsFocused(isFocused);
+      updateFocus(isFocused, headerEl);
+    });
+  }, [index, onFocusUpdate]);
+
+  useEffect(() => {
+    updateFocus(isFocused, headerEl);
+  }, [headerEl]);
```

Comment:
```suggestion
  useEffect(() => {
    onFocusUpdate([index, -1], (isFocused: boolean) => {
      setIsFocused(isFocused);

      if (isFocused && headerEl) {
        // Only focus the cell if not already focused on something in the cell
        if (!headerEl.contains(document.activeElement)) {
          headerEl.focus();
        }
      }
    });
  }, [index, onFocusUpdate, headerEl]);
```

Per Zoom chat - remove 2nd useEffect and extra fn, just inline the `.focus()` logic in the callback